vrx_gazebo/include/vrx_gazebo/ball_shooter_plugin.hh

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef VRX_GAZEBO_BALL_SHOOTER_PLUGIN_HH_
+#define VRX_GAZEBO_BALL_SHOOTER_PLUGIN_HH_
+
+#include <ros/ros.h>
+#include <std_msgs/Empty.h>
+#include <memory>
+#include <mutex>
+#include <gazebo/gazebo.hh>
+#include <sdf/sdf.hh>
+
+namespace gazebo
+{
+/// \brief Simulate a ball shooter. A projectile is launched when a
+/// ROS message is received. The ball is reused (teleported) from previous
+/// shots.
+///
+/// The plugin accepts the following SDF parameters:
+/// * Required parameters:
+/// <projectile> The object used as projectile should be declared as a
+///              <projectile> element with the following parameters:
+///                 * <model_name> The name of the model used as projectile.
+///                                Required parameter.
+///                 * <link_name> The name of the link within the projectile
+///                               model where the force will be applied when
+///                               firing the shooter. Required parameter.
+///                 * <frame> If present, the <pose> parameter will be in the
+///                           frame of this link/model. Otherwise the world
+///                           frame is used. Optional parameter.
+///                 * <pose> - Pose of the projectile right before being shot.
+///                            Optional parameter. Default to {0 0 0 0 0 0}.
+///
+/// * Optional parameters:
+/// <num_shots> - Number of shots allowed. Default to UINT_MAX.
+/// <shot_force> - Force (N) applied to the projectile. Default to 250 N.
```

Comment:
What do you think about including this tag explicitly in the example and also the .xacro file we are providing ? Out of all the parameters, it seems like the one that users are most likely to want to adjust. Even if it is redundantly set to 250 it might make it easier to quickly pick up how to use this plugin.

Reply:
Good idea! Added in 07609f6.